keras/callbacks.py

```diff
@@ -214,10 +253,7 @@ class BaseLogger(Callback):
     """
 
     def __init__(self, stateful_metrics=None):
-        if stateful_metrics:
-            self.stateful_metrics = set(stateful_metrics)
-        else:
-            self.stateful_metrics = set()
```

Comment:
Could you revert the changes unrelated to the sync of the callback api with tf.keras? You can create another pull request for them if you want. It's to make it easier to review and discuss each of your changes. Thanks.